### PR TITLE
[WIP] Sync env vars between dokku and ic

### PR DESCRIPTION
**Pull request by brambokdam**
This PR adds a create button on the ENV var page that allows you to sync the env vars in the dokku app, with the IC app.

Fixes: https://github.com/intercity/intercity-next/issues/47

screen shot 2016-05-13 at 10 04 17

### CODE REVIEW

File: app/controllers/env_vars_controller.rb [hunk 1]
```diff
@@ -16,6 +16,11 @@ def destroy
     DeleteEnvVarJob.perform_later(@app, @env_var.key) if @env_var.destroy
   end
 
+  def sync
```
Comment (jvanbaarsen) [inline app/controllers/env_vars_controller.rb]: Can we move this to its own controller so we stick to the REST actions?

File: app/jobs/sync_env_vars_job.rb [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+    SshExecution.new(app.server).execute_with_block do |ssh|
+      ssh.open_channel do |channel|
+        channel.exec "dokku config #{app.clean_name}" do |exec_channel, success|
+          unless success
+            logger.error "Could not update env vars"
+            abort "could not update env vars"
+          end
+          exec_channel.on_data do |_, data|
+            next if data.starts_with?("=====>")
+            key, value = data.split(":").map(&:strip)
```
Comment (jvanbaarsen) [inline app/jobs/sync_env_vars_job.rb]: What happens if I add a envvar which contains a `:`?

File: app/views/env_vars/index.html.erb [hunk 3]
```diff
@@ -1,6 +1,10 @@
 <%= render "apps/heading", app: @app, tab: "env_vars" %>
 <div class="columns">
   <div class="column is-8">
+    <div class="notification is-success is-hidden" data-role="sync-notification">
```
Comment (jvanbaarsen) [inline app/views/env_vars/index.html.erb]: Do we also automaticly refresh once those env vars are added? Since this message indicate we do

**Comment by jvanbaarsen** [comment]
Can we get some tests going for this new behaviour?

**Comment by jvanbaarsen** [comment]
Also I'm not sure if #48 is getting fixed by this PR?

**Comment by brambokdam** [comment]
Nope your right had to be #47

**Comment by michiels** [comment]
@brambokdam UI suggestion: Make the synchronize button on its own widget (above/below the add widget) with a single line of explanation that ENV vars will be fetched from the server). Or you could add that button and such a message below/above the ENV vars table.

The button location currently feels a bit illogical and you don't really know what it means.

**Comment by jvanbaarsen** [comment]
:+1: For Michiels suggestion

File: app/jobs/sync_env_vars_job.rb (hunk 2 above)
Comment (brambokdam) [inline app/jobs/sync_env_vars_job.rb]: For some reason you can't add a variable with `:` in it.

**Comment by brambokdam** [comment]
+1 For the UI suggestion looks a lot better 👍

**Comment by michiels** [comment]
@brambokdam cool! what does syncrhonize do? Does it only fetch "new" variables from your server and add them to IC?

File: app/jobs/sync_env_vars_job.rb (hunk 2 above)
Comment (jvanbaarsen) [inline app/jobs/sync_env_vars_job.rb]: are you sure? I believe you can. Let me check

**Comment by jvanbaarsen** [comment]
@brambokdam How do I know if the vars are out of sync? The text in the UI indicates to me that there is a way to see this.

**Comment by brambokdam** [comment]
@jvanbaarsen I don't agree "you can if they are out of sync" but if you have a better suggestion I'm all ears. 👍

**Comment by jvanbaarsen** [comment]
@brambokdam Thats the thing, I don't know a better way xD I just think this copy is maybe not as good as it can be. @michiels Suggestions?

**Comment by jvanbaarsen** [comment]
Closing this for now. Thanks for the work though.

**Pull request closed without merging**